### PR TITLE
refactor(IDX): move docker login out of before-script

### PR DESCRIPTION
**Pull request by cgundy**
This PR moves docker login out of the before-script and into its own step, so it will only be used where it is actually needed.

### CODE REVIEW

File: .github/workflows-source/ci-pr-only.yml [hunk 1]
```diff
@@ -38,13 +37,6 @@ anchors:
       <<: *image
       options: >-
         -e NODE_NAME
-  before-script: &before-script
```
Comment (marko-k0) [inline .github/workflows-source/ci-pr-only.yml:41]: I think it's beneficial if we get node info on all CI jobs that we run on on-prem runners, so I'd retain before-script in CI jobs of this workflow.

File: .github/workflows-source/ci-main.yml [hunk 2]
```diff
@@ -61,9 +60,15 @@ anchors:
     name: Before script
     id: before-script
     shell: bash
-    run: ./gitlab-ci/src/ci-scripts/before-script.sh
-    env:
-      DOCKER_HUB_PASSWORD_RO: ${{ secrets.DOCKER_HUB_PASSWORD_RO }}
+    run: |
```
Comment (marko-k0) [inline .github/workflows-source/ci-main.yml:63]: This **might** be a regression. In the script we use now we login 3 times. Note that we still have `docker` invocation essentially calling `sudo podman $@`. That means we login with `sudo podman login`, we login with docker with current CI user (`docker-bin login`) and we login with docker with root (`sudo docker-bin login`). Credentials in all 3 cases end up being saved in different places. I think rootful podman might actually use credentials from docker (root user) if they're available so that might be unnecessary. This is very convoluted but we went with "better safe then sorry" approach. I believe we do have `sudo podman` being used while building IC-OS images (implicitly probably pulling some image from dockerhub) so we'd need to have `/root/.docker/config.json` available (assuming rootful podman picks it) and with this change, we probably need to copy it from current's user home to root's home.

In short, I'd be happy to welcome this change, but it's risky and we'd need to dig into our IC-OS to see if we can allow ourselves to do this. Docker rate limit issues only surface when we exceed number of requests we get with anonymous access.

Comment (marko-k0) [reply .github/workflows-source/ci-main.yml:63]: Actually, now that we moved our base images to GHCR, it might be more feasible. I'll sync with node team.

Comment (cgundy) [reply .github/workflows-source/ci-main.yml:63]: Yeah that's a fair point. I can keep the original docker login script we have, but still make my change to only use it when necessary to avoid exposing secrets without needing to. But I think we should still look into cleaning this up in the future, as it is quite convoluted.

Comment (marko-k0) [reply .github/workflows-source/ci-main.yml:63]: Yes, let's do that and sync tomorrow regarding the future of this and the use of GHCR/DockerHub.

**Comment by marko-k0** [comment]
I see `schedule-weeky.yml` is using `before-script.sh` as well.

**Comment by cgundy** [comment]
> I see schedule-weeky.yml is using before-script.sh as well. And also system-tests-k8s.yml

File: .github/workflows/system-tests-k8s.yml [hunk 3]
```diff
@@ -80,6 +78,8 @@ jobs:
       - name: Run System Tests on K8s
         id: bazel-test-all
         uses: ./.github/actions/bazel-test-all/
+        env:
```
Comment (marko-k0) [inline .github/workflows/system-tests-k8s.yml:81]: It's the other way around. We don't publish any artifacts so we don't need AWS credentials but good to leave docker-login logic as container images in docker rules, do get pulled.

**Review by marko-k0 — APPROVED** [review]
